aiosysbus/access.py

```diff
@@ -143,33 +158,41 @@ def _perform_request(self, verb, **kwargs):
 
         return resp["result"] if "result" in resp else None
 
-    def get(self, service, method, parameters={}):
+    def get(
+        self, service: str, method: str, parameters: dict[str, Any] | None = {}
+    ) -> dict[str, Any] | None:
         """Send get request and return results."""
         data = {"service": service, "method": method, "parameters": parameters}
         return self._perform_request(self.session.get, json=data)
 
-    def post(self, service, method, parameters={}):
+    def post(
+        self, service: str, method: str, parameters: dict[str, Any] | None = {}
+    ) -> dict[str, Any] | None:
         """Send post request and return results."""
         data = {"service": service, "method": method, "parameters": parameters}
         return self._perform_request(self.session.post, json=data)
 
-    def put(self, service, method, parameters={}):
+    def put(
+        self, service: str, method: str, parameters: dict[str, Any] | None = {}
+    ) -> dict[str, Any] | None:
         """Send put request and return results."""
         data = {"service": service, "method": method, "parameters": parameters}
         return self._perform_request(self.session.put, json=data)
 
-    def delete(self, service, method, parameters={}):
+    def delete(
+        self, service: str, method: str, parameters: dict[str, Any] | None = {}
+    ) -> dict[str, Any] | None:
         """Send delete request and return results."""
         data = {"service": service, "method": method, "parameters": parameters}
         return self._perform_request(self.session.delete, json=data)
 
-    def get_permissions(self):
+    def get_permissions(self) -> str:
         """Return the permissions for this session/app."""
         if not self.session_permissions:
             self._refresh_session_token()
-        return self.session_permissions
+        return self.session_permissions  # type: ignore[return-value]
```

Comment:
For now I added `type: ignore` to keep existing behavior, but maybe it should allow `None` return?

Reply:
yes the return can be a string or None